src/libs/TravelUtils/index.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * Determines whether the given URL points to an Expensify Travel (Trip) page.
+ *
+ * A link is considered a travel link if:
+ * - it belongs to any `*.travel.expensify.com` domain, or
+ * - it belongs to any `*.expensify.com` domain and contains a `/trips/` path segment
+ *
+ * The check is case-insensitive and safely handles invalid URLs.
+ *
+ * @param url - URL string to evaluate
+ * @returns true if the URL is an Expensify Travel / Trip link, otherwise false
+ */
+function isTravelLink(url: string): boolean {
+    if (!url) {
+        return false;
+    }
+
+    try {
+        const {hostname, pathname} = new URL(url.toLowerCase());
+
+        // Any travel subdomain (prod, staging, dev, etc.)
+        if (hostname.includes('travel.expensify.com')) {
```

Comment:
Lets make it stricter spefically ensures that the domain only ends with the travel domain:

```suggestion
        if (hostname.endsWith('travel.expensify.com')) {
```